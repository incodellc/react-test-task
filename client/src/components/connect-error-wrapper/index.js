import ConnectErrorWrapper from './ConnectErrorWrapper';
export default ConnectErrorWrapper;