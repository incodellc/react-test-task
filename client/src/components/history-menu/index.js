import HistoryMenu from './HistoryMenu';
export default HistoryMenu;