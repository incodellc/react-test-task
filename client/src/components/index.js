import InfoBar from './info-bar';
import Layout from './layout';
import ConnectErrorWrapper from './connect-error-wrapper';
import TickerList from './ticker-list';
import TickerListItem from './ticker-list-item';

export { InfoBar, Layout, ConnectErrorWrapper, TickerList, TickerListItem };