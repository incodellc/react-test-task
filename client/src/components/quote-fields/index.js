import QuoteFields from './QuoteFields';
export default QuoteFields;