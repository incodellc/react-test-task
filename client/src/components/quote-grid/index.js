import QuotesGrid from './QuotesGrid';
export default QuotesGrid;