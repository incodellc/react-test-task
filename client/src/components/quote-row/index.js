import QuoteRow from './QuoteRow';
export default QuoteRow;