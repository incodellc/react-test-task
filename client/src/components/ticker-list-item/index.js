import TickerListItem from './TickerListItem';
export default TickerListItem;