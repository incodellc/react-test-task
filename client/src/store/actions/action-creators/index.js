import actionTypes from '../action-types';
import { tickerService } from '../../../services';
import { tickers } from '../../../utils';

export const streamServerConnect = () => {
    return {
        type: actionTypes.STREAM_SERVER_CONNECT
    };
};

export const streamServerDisconnect = () => {
    return {
        type: actionTypes.STREAM_SERVER_DISCONNECT
    };
};

export const updateQuotes = (data) => {
    return {
        type: actionTypes.UPDATE_QUOTES,
        data
    };
};

export const connectToStreamServer = () => {
    return dispatch => {
        dispatch({
            type: actionTypes.CONNECTION_INIT
        });
        tickerService(
            tickers, 
            () => dispatch(streamServerConnect()), 
            (data) => dispatch(updateQuotes(data)),
            () => dispatch(streamServerDisconnect())
        );
    };
};