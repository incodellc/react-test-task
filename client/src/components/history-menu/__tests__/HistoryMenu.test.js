import React from 'react';
import { quotesExample } from '../../../utils';
import { renderWithRedux } from '../../../utils/test-utils';
import { HistoryMenu } from '../../../components';
import { resetHistory, resetSortParams, updateQuotes, applySortParams } from '../../../store/actions/action-creators';
import { cleanup, fireEvent } from '@testing-library/react';

describe('Check HistoryMenu component', () => {
    afterEach(cleanup);

    it('component renders correctly', () => {
        const { asFragment } = renderWithRedux(<HistoryMenu />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('component should contain expected elements and text content', () => {
        const { getByTestId } = renderWithRedux(<HistoryMenu />);
        expect(getByTestId('history-menu')).toContainElement(getByTestId('history-menu-left'));
        expect(getByTestId('history-menu')).toContainElement(getByTestId('history-menu-right'));
        expect(getByTestId('history-menu')).toContainElement(getByTestId('history-menu-dropdown'));
        expect(getByTestId('history-menu')).toContainElement(getByTestId('history-menu-reset-history'));
        expect(getByTestId('history-menu')).toContainElement(getByTestId('history-menu-reset-sorting'));
        expect(getByTestId('history-menu-left')).toHaveTextContent('Quotes shown:');
        expect(getByTestId('history-menu-left')).toHaveTextContent('10 out of 10');
        expect(getByTestId('history-menu-right')).toHaveTextContent('Max history length:');
    });
    it('check max history dropdown content an behavior', () => {
        const { getByRole, getAllByRole, store } = renderWithRedux(<HistoryMenu />);
        store.getState().historyLengthOptions.forEach((element, index) => {
            expect(getAllByRole('option')[index]).toHaveTextContent(element.text);
        });
        expect(getByRole('alert')).toHaveTextContent('100 quotes');
        fireEvent.click(getAllByRole('option')[1]);
        expect(getByRole('alert')).toHaveTextContent('200 quotes');
        fireEvent.click(getAllByRole('option')[0]);
        expect(getByRole('alert')).toHaveTextContent('100 quotes');
    });
    it('check reset history menu item content an behavior', () => {
        const { getByTestId, store } = renderWithRedux(<HistoryMenu />);
        expect(getByTestId('history-menu-reset-history')).toHaveTextContent('Reset History');
        expect(getByTestId('history-menu-reset-history')).not.toHaveClass('disabled');
        store.dispatch(resetHistory());
        expect(getByTestId('history-menu-reset-history')).toHaveTextContent('Reset History');
        expect(getByTestId('history-menu-reset-history')).toHaveClass('disabled');
        store.dispatch(updateQuotes(quotesExample[0]));
        expect(getByTestId('history-menu-reset-history')).toHaveTextContent('Reset History');
        expect(getByTestId('history-menu-reset-history')).not.toHaveClass('disabled');
    });
    it('check reset sort params menu item content and behavior', () => {
        const { getByTestId, store } = renderWithRedux(<HistoryMenu />);
        expect(getByTestId('history-menu-reset-sorting')).toHaveTextContent('Reset Sorting');
        expect(getByTestId('history-menu-reset-sorting')).toHaveClass('disabled');
        store.dispatch(applySortParams('ticker'));
        expect(getByTestId('history-menu-reset-sorting')).toHaveTextContent('Reset Sorting');
        expect(getByTestId('history-menu-reset-sorting')).not.toHaveClass('disabled');
        store.dispatch(resetSortParams());
        expect(getByTestId('history-menu-reset-sorting')).toHaveTextContent('Reset Sorting');
        expect(getByTestId('history-menu-reset-sorting')).toHaveClass('disabled');
    });
});