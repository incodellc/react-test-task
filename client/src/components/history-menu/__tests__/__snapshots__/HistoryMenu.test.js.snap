// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check HistoryMenu component component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="ui text historyMenu menu"
    data-testid="history-menu"
  >
    <div
      class="left menu"
      data-testid="history-menu-left"
    >
      <div
        class="item"
      >
        <span>
          Quotes shown:
        </span>
        <b>
          10 out of 10
        </b>
      </div>
    </div>
    <div
      class="right menu"
      data-testid="history-menu-right"
    >
      <div
        class="item"
      >
        <span>
          Max history length:
        </span>
        <div
          aria-expanded="false"
          class="ui inline scrolling dropdown"
          data-testid="history-menu-dropdown"
          role="listbox"
          tabindex="0"
        >
          <div
            aria-atomic="true"
            aria-live="polite"
            class="text"
            role="alert"
          >
            100 quotes
          </div>
          <i
            aria-hidden="true"
            class="dropdown icon"
          />
          <div
            class="menu transition"
          >
            <div
              aria-checked="true"
              aria-selected="true"
              class="active selected item"
              role="option"
              style="pointer-events: all;"
            >
              <span
                class="text"
              >
                100 quotes
              </span>
            </div>
            <div
              aria-checked="false"
              aria-selected="false"
              class="item"
              role="option"
              style="pointer-events: all;"
            >
              <span
                class="text"
              >
                200 quotes
              </span>
            </div>
            <div
              aria-checked="false"
              aria-selected="false"
              class="item"
              role="option"
              style="pointer-events: all;"
            >
              <span
                class="text"
              >
                500 quotes
              </span>
            </div>
            <div
              aria-checked="false"
              aria-selected="false"
              class="item"
              role="option"
              style="pointer-events: all;"
            >
              <span
                class="text"
              >
                1000 quotes
              </span>
            </div>
          </div>
        </div>
      </div>
      <a
        class="item"
        data-testid="history-menu-reset-history"
      >
        Reset History
      </a>
      <a
        class="disabled item"
        data-testid="history-menu-reset-sorting"
      >
        Reset Sorting
      </a>
    </div>
  </div>
</DocumentFragment>
`;
